drivers/flash/flash_stm32_ospi.c

```diff
@@ -0,0 +1,1498 @@
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* Enables writing to the memory sending a Write Enable and wait it is effective */
+static int stm32_ospi_write_enable(OSPI_HandleTypeDef *hospi, uint8_t nor_mode, uint8_t nor_rate)
+{
+	OSPI_AutoPollingTypeDef s_config = {0};
+	OSPI_RegularCmdTypeDef s_command = ospi_prepare_cmd(nor_mode, nor_rate);
+
+	/* Initialize the write enable command */
+	s_command.Instruction = (nor_mode == OSPI_SPI_MODE)
+				? SPI_NOR_CMD_WREN
+				: SPI_NOR_OCMD_WREN;
+	s_command.AddressMode = HAL_OSPI_ADDRESS_NONE;
+	s_command.DataMode    = HAL_OSPI_DATA_NONE;
+	s_command.DummyCycles = 0U;
+
+	/* Send the command */
+	if (HAL_OSPI_Command(hospi, &s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI flash write enable cmd failed");
+		return -EIO;
+	}
+
+	/* Configure automatic polling mode to wait for write enabling */
+	s_command.Instruction    = (nor_mode == OSPI_SPI_MODE)
+				? SPI_NOR_CMD_RDSR
+				: SPI_NOR_OCMD_RDSR;
+	s_command.AddressMode    = (nor_mode == OSPI_SPI_MODE)
+				? HAL_OSPI_ADDRESS_1_LINE
+				: HAL_OSPI_ADDRESS_8_LINES;
+	s_command.Address        = 0U;
+	s_command.DummyCycles    = (nor_mode == OSPI_SPI_MODE)
+				? 0U
+				: ((nor_rate == OSPI_DTR_TRANSFER)
+				? SPI_NOR_DUMMY_REG_OCTAL_DTR
+				: SPI_NOR_DUMMY_REG_OCTAL);
+	s_command.DataMode       = (nor_mode == OSPI_SPI_MODE)
+				? HAL_OSPI_DATA_1_LINE
+				: HAL_OSPI_DATA_8_LINES;
+	s_command.NbData         = (nor_rate == OSPI_DTR_TRANSFER) ? 2U : 1U;
+
+	/* Send the command */
+	if (HAL_OSPI_Command(hospi, &s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI config auto polling cmd failed");
+		return -EIO;
+	}
+
+	s_config.Match           = SPI_NOR_WREN_MATCH;
+	s_config.Mask            = SPI_NOR_WREN_MASK;
+	s_config.MatchMode       = HAL_OSPI_MATCH_MODE_AND;
+	s_config.Interval        = SPI_NOR_AUTO_POLLING_INTERVAL;
+	s_config.AutomaticStop   = HAL_OSPI_AUTOMATIC_STOP_ENABLE;
+
+	if (HAL_OSPI_AutoPolling(hospi, &s_config, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI config auto polling failed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* Write Flash configuration register 2 with new dummy cycles */
+static int stm32_ospi_write_cfg2reg_dummy(OSPI_HandleTypeDef *hospi,
+					uint8_t nor_mode, uint8_t nor_rate)
+{
+	OSPI_RegularCmdTypeDef s_command = s_command = ospi_prepare_cmd(nor_mode, nor_rate);
+
+	/* Initialize the writing of configuration register 2 */
+	s_command.Instruction = (nor_mode == OSPI_SPI_MODE)
+				? SPI_NOR_CMD_WR_CFGREG2
+				: SPI_NOR_OCMD_WR_CFGREG2;
+	s_command.Address = SPI_NOR_REG2_ADDR3;
+	s_command.DummyCycles = 0U;
+	s_command.NbData = (nor_mode == OSPI_SPI_MODE) ? 1U
+			: ((nor_rate == OSPI_DTR_TRANSFER) ? 2U : 1U);
+
+	/* Send the command */
+	if (HAL_OSPI_Command(hospi, &s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI transmit ");
+		return -EIO;
+	}
+	uint8_t tmp = SPI_NOR_CR2_DUMMY_CYCLES_66MHZ;
```

Comment:
Could that be moved to the top of the function and named in a way to add some meaning to it (i.e. `dummy_cycles`)?